File: app/src/main/java/dolphin/android/net/HttpHelper.java
package dolphin.android.net;

import android.content.Context;
import android.net.ConnectivityManager;

/**
 * Created by dolphin on 2013/6/3.
 */
public class HttpHelper extends UrlHelper {
    private static final String TAG = "HttpHelper";

    /**
     * HTTP status code when no server error has occurred.
     */
    public static final int HTTP_STATUS_OK = 200;

    /**
     * check if network is available
     *
     * @param context Context
     * @return true if network available
     */
    public static boolean checkNetworkAvailable(Context context) {
        ConnectivityManager connMgr =
                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        if (connMgr != null && connMgr.getActiveNetworkInfo() != null) {
            return connMgr.getActiveNetworkInfo().isAvailable();
        }
        return false;
    }

    /**
     * check if network is connected
     *
     * @param context Context
     * @return true if network connected
     */
    public static boolean checkNetworkConnected(Context context) {
        ConnectivityManager connMgr =
                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        if (connMgr != null && connMgr.getActiveNetworkInfo() != null) {
            return connMgr.getActiveNetworkInfo().isConnected();
        }
        return false;
    }

}

File: app/src/main/java/dolphin/android/net/HttpHelper.kt
package dolphin.android.net

import android.content.Context
import android.net.ConnectivityManager

/**
 * Created by dolphin on 2013/6/3.
 */
open class HttpHelper(val context: Context) : UrlHelper() {
    companion object {
        private const val TAG = "HttpHelper"

        /**
         * HTTP status code when no server error has occurred.
         */
        const val HTTP_STATUS_OK = 200

        /**
         * check if network is available
         *
         * @param context Context
         * @return true if network available
         */
        @JvmStatic
        fun checkNetworkAvailable(context: Context): Boolean {
            val connMgr = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            return if (connMgr.activeNetworkInfo != null) {
                connMgr.activeNetworkInfo.isAvailable
            } else false
        }

        /**
         * check if network is connected
         *
         * @param context Context
         * @return true if network connected
         */
        @JvmStatic
        fun checkNetworkConnected(context: Context): Boolean {
            val connMgr = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            return if (connMgr.activeNetworkInfo != null) {
                connMgr.activeNetworkInfo.isConnected
            } else false
        }
    }

    val isNetworkAvailable: Boolean
        get() = (context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager)
                .activeNetworkInfo?.isAvailable ?: false

    val isNetworkConnected: Boolean
        get() = (context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager)
                .activeNetworkInfo?.isConnected ?: false

    fun getRemoteFile(url: String, path: String, listener: HttpProgressListener? = null): Boolean {
        return UrlHelper.getRemoteFile(context, url, path, DEFAULT_NETWORK_TIMEOUT, listener)
    }
}
